Client/mods/deathmatch/logic/CClientEntity.cpp

```diff
@@ -1070,6 +1070,29 @@ bool CClientEntity::IsEntityAttached(CClientEntity* pEntity)
     return ListContains(m_AttachedEntities, pEntity);
 }
 
+bool CClientEntity::IsAttachedToElement(CClientEntity* pEntity, bool bRecursive)
+{
+    if (bRecursive)
+    {
+        std::set<CClientEntity*> history;
+
+        for (CClientEntity* pCurrent = this; pCurrent; pCurrent = pCurrent->GetAttachedTo())
+        {
+            if (pCurrent == pEntity)
+                return true;
+
+            if (MapContains(history, pCurrent))
+                break; // This should not be possible, but you never know
+
+            MapInsert(history, pCurrent);
```

Comment:
Why not just use the return value of set::insert? It returns false if theres already a matching element in the set. Much faster than checking, then inserting.
Like this: 
```cpp
if (!std::get<bool>(history.insert(pCurrent)); 
    break; // This should not be possible, but you never know
```

Reply:
It is just a copy-paste from the old code

Reply:
Yeah, I though, but its probably like a decade old, now there's a better way to do it :D